test/unit/index.js

```diff
@@ -188,3 +183,37 @@ test('make manifest available to other webpack plugins', async (t) => {
     t.pass();
   }
 });
+
+if (!webpack.version.startsWith('4')) {
+  test('works with asset modules', async (t) => {
+    const config = {
+      context: __dirname,
+      entry: '../fixtures/import_image.js',
+      output: {
+        path: join(outputPath, 'auxiliary-assets'),
+        assetModuleFilename: `images/[name].[hash:4][ext]`
+      },
+      module: {
+        rules: [
+          {
+            test: /\.(svg)/,
+            type: 'asset/resource'
+          }
+        ]
+      }
+    };
+
+    const { manifest } = await compile(config, t);
+    const expected = {
+      'main.js': 'main.js',
+      'images/manifest.svg': `images/manifest.14ca.svg`
+    };
+
+    t.truthy(manifest);
+    t.deepEqual(Object.keys(expected), ['main.js', 'images/manifest.svg']);
+    t.deepEqual(manifest['main.js'], 'main.js');
+    t.regex(manifest['images/manifest.svg'], /images\/manifest\.[a-z|\d]{4}\.svg/);
```

Comment:
Doing this so I could do a fuzzy check on the hash. The `[fullhash]` thing does not work for `assetModuleFilename`, so I needed to use a *real* hash.